rest/model/task_annotations.go

```diff
@@ -3,57 +3,50 @@ package model
 import (
 	"time"
 
-	"github.com/evergreen-ci/birch"
 	"github.com/evergreen-ci/evergreen/model/task_annotations"
 )
 
 type APITaskAnnotation struct {
-	Id              *string             `bson:"_id" json:"id"`
-	TaskId          *string             `bson:"task_id" json:"task_id"`
-	TaskExecution   int                 `bson:"task_execution" json:"task_execution"`
-	Note            *string             `bson:"note" json:"note"`
-	Issues          []APIIssueLink      `bson:"issues" json:"issues"`
-	SuspectedIssues []APIIssueLink      `bson:"suspected_issues" json:"suspected_issues"`
-	Source          APIAnnotationSource `bson:"source" json:"source"`
-	Metadata        *birch.Document     `bson:"metadata" json:"metadata"`
+	Id             *string       `bson:"_id" json:"id"`
+	TaskId         *string       `bson:"task_id" json:"task_id"`
+	TaskExecution  int           `bson:"task_execution" json:"task_execution"`
+	APIAnnotation  APIAnnotation `bson:"api_annotation,omitempty" json:"api_annotation,omitempty"`
+	UserAnnotation APIAnnotation `bson:"user_annotation,omitempty" json:"user_annotation,omitempty"`
 }
 
+type APIAnnotation struct {
+	Note          *string             `bson:"note,omitempty" json:"note,omitempty"`
+	Issues        []APIIssueLink      `bson:"issues,omitempty" json:"issues,omitempty"`
+	SuspectIssues []APIIssueLink      `bson:"suspected_issues,omitempty" json:"suspected_issues,omitempty"`
+	Source        APIAnnotationSource `bson:"source" json:"source"`
+}
 type APIAnnotationSource struct {
-	Author *string    `json:"author"`
-	Time   *time.Time `json:"time"`
+	Author *string    `bson:"author,omitempty" json:"author,omitempty"`
+	Time   *time.Time `bson:"time,omitempty" json:"time,omitempty"`
 }
+
 type APIIssueLink struct {
-	URL      *string `json:"url"`
-	IssueKey *string `json:"issue_key"`
+	URL      *string `bson:"url" json:"url"`
```

Comment:
will URL not ever be empty?

Reply:
I don't think it would ever make sense for someone to give us an issue that has an IssueKey (the display text) but no url. Should I add omitempty just to be sure?

Reply:
might as well 🤷